pkg/costmodel/costmodel.go

```diff
@@ -1687,6 +1729,19 @@ func (cm *CostModel) costDataRange(cli prometheusClient.Client, cp costAnalyzerC
 		klog.V(1).Infof("Unable to get Pod Labels for Metrics: %s", err.Error())
 	}
 
+	nsAnnotations, err := GetNamespaceAnnotationsMetrics(resNSAnnotations, clusterID)
+	if err != nil {
+		klog.V(1).Infof("Unable to get Namespace Annotations for Metrics: %s", err.Error())
+	}
+	if nsAnnotations != nil {
+		appendNamespaceLabels(namespaceAnnotationsMapping, nsAnnotations)
```

Comment:
We can probably rename `appendNamespaceLabels` to just `mergeStringMap` or something :)